### PR TITLE
Add simulated event queue

### PR DESCRIPTION
**Pull request by yuriy-arabskyy**

### CODE REVIEW

File: src/qubx/backtester/simulator.py [hunk 1]
```diff
@@ -386,79 +350,111 @@ def subscribe(
...
+        qiter = self._data_queue.create_iterator(start, end)
+        total_duration = pd.Timestamp(end) - pd.Timestamp(start)
+        prev_dt = pd.Timestamp(start)
+
+        with tqdm(total=total_duration.total_seconds(), desc="Simulating", unit="s", leave=False) as pbar:
+            for symbol, event in tqdm(qiter):
+                _run(symbol, event)
+                dt = pd.Timestamp(event.time)
+                # update only if date has changed
+                if dt - prev_dt > pd.Timedelta("1D"):
```
Comment (yuriy-arabskyy) [inline src/qubx/backtester/simulator.py:408]: I will make this dynamic tmr

File: src/qubx/data/readers.py [hunk 2]
```diff
@@ -837,38 +860,61 @@ def _read(
...
-        transform.start_transform(data_id, names, start=start, stop=stop)
+        if chunksize > 0:
 
-        transform.process_data(records)
-        return transform.collect()
+            def _iter_chunks():
+                while True:
+                    records = _cursor.fetchmany(chunksize)
+                    if not records:
+                        _cursor.close()
```
Comment (yuriy-arabskyy) [inline src/qubx/data/readers.py:881]: I can probably rewrite this better

Comment (dmarienko) [reply src/qubx/data/readers.py:881]: Looks good as for me

**Review by dmarienko — APPROVED** [review]
I liked this idea on SimulatedDataQueue !

**Pull request merged**